docs/vm-metrics.md

```diff
@@ -0,0 +1,254 @@
+# Overview
+
+This document will cover
+
+- what metrics are relevant for VMs, highlighting overlaps and additions
+  with respect to Kubernetes core metrics
+- node agent: collection of the VM metrics
+- implementation of the monitoring pipeline: extra components besides the node agent
+
+## Libvirt metrics
+[libvirt](https://libvirt.org/) is the virtualization framework that
+KubeVirt uses. The libvirt control daemon, libvirtd, does its own independent
+collection of the VM metrics. 
+libvirtd uses both the QEMU configuration and the standard
+facilites of the linux kernel to do the resource limitation and
+accounting. For example, libvirt uses cgroups, to do resource
+accounting for memory, cpu usage, I/O.
+
+Libvirtd provides metrics that roughly overlaps with Kubernetes
+[core metrics](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/monitoring_architecture.md)
```

Comment:
Looks like in that proposal, the core metrics should not be extensible by design, so we can't add core metrics which might be relevant for VMs only there?